src/c++/perf_analyzer/concurrency_manager.cc

```diff
@@ -151,6 +151,7 @@ void
 ConcurrencyManager::ResumeSequenceWorkers()
 {
   if (on_sequence_model_) {
+    UnpauseAllSequences();
```

Comment:
Why'd this move from HandleExecuteOff() to here? Can you explain how these are equivalent or why a design change was made?

Reply:
Concurrency worker/manager currently has some nastiness around ownership of sequences.  It is a formula based on other thread's concurrencies. If thread 0 has concurrency of 2 and thread 1 has concurrency of 2, then thread 0 owns sequence slots 0 and 1, and thread 1 owns sequence slots 2 and 3.  If we then update both to have concurrency of 3, then thread 0 owns sequence slots 0,1,2, while thread 1 owns slots 3,4,5. As you can see, seq slot 2 changed ownership when the concurrency was updated.

I don't remember the exact race I ran into, but it was something around sequences pausing their old set of owned sequences, but then unpausing the new set. There was some kind of race that caused not all sequences to get unpaused and a hang resulted.  I felt it was cleaner to make sure everyone was unpaused in one clean location.

Reply:
That is terrible... Thanks for catching.